enterprise/api/v1/serializers.py

```diff
@@ -190,6 +198,47 @@ def get_enterprise_customer_catalogs(self, obj):
         """
         return [str(catalog.uuid) for catalog in obj.enterprise_customer_catalogs.all()]
 
+    def get_enterprise_notification_banner(self, obj):
+        """
+        Return the notification text if exist OR None
+        """
+
+        try:
+            # this serializer is also called from tpa_pipeline and request is not available in the first place there.
+            request = self.context['request']
+            user_id = request.user.id
+        except Exception as exc:  # pylint: disable=broad-except
+            LOGGER.error('[Admin Notification API] Get enterprise notification banner request object not found,'
+                         ' Enterprise Customer :{} Exception: {}'.format(obj.slug, exc))
+            return None
+        now = datetime.datetime.now(pytz.UTC)
+        read = True
+        notification_text = None
+
+        notification = AdminNotification.objects.filter(start_date__lte=now, expiration_date__gte=now,
+                                                        is_active=True).first()
+        if notification:
+            try:
+                # verify that user didn't read the notification
+                enterprise_customer_user = EnterpriseCustomerUser.objects.get(enterprise_customer=obj,
+                                                                              user_id=user_id)
+                read = AdminNotificationRead.objects.filter(enterprise_customer_user=enterprise_customer_user,
+                                                            is_read=True, admin_notification=notification).exists()
+            except (TypeError, ValueError, EnterpriseCustomerUser.DoesNotExist):
+                error_message = ('[Admin Notification API] EnterpriseCustomerUser not exist for User: {}, '
+                                 ' EnterpriseCustomer:{}').format(user_id, obj.slug)
+                LOGGER.error(error_message)
+
+        if not read:
+            notification_text = notification.text
+            filters = notification.admin_notification_filter.all()
+            for notification_filter in filters:
+                # if filter didn't match to filters in enterprise_customer OR filter is not checked in
+                # enterprise_customer then return None
+                if not getattr(obj, notification_filter.filter, None):
+                    notification_text = None
+        return notification_text
```

Comment:
I think you should  also return other information related to notification like `notification_id`, `is_read` etc.

Reply:
At least `notification_id` should be included as that is needed in the frontend when marking a notification as read.